src/workerd/util/mimetype.c++

```diff
@@ -378,4 +383,64 @@ bool MimeType::isVideo(const MimeType& mimeType) {
 bool MimeType::isAudio(const MimeType& mimeType) {
   return mimeType.type() == "audio";
 }
+
+kj::Maybe<MimeType> MimeType::extract(kj::StringPtr input) {
+  kj::Maybe<MimeType> mimeType;
+
+  constexpr static auto findNextSeparator = [](auto& input) -> kj::Maybe<size_t> {
+    // Scans input to find the next comma (,) that is not contained within
+    // a quoted section, returning the position of the comma or kj::none
+    // if not found.
+    for (size_t i = 0; i < input.size(); ++i) {
+      if (input[i] == '"') {
+        // Skip to the end of the quoted section
+        while (++i < input.size() && (input[i] != '"' || input[i-1] == '\\')) {}
```

Comment:
I'm confused by the handling of inputs like `\"foobar` here. Doesn't that indicate that it's not a quoted section in which case nothing should be skipped? But this code as it is right now will skip to the end of the "foobar", am I misreading something or is this incorrect?

Reply:
ah, likely need an additional condition on the previous line to ensure the `"` we're on currently isn't escaped. Will update and add a test (that edge case is apparently not covered by the existing web platform tests)